src/hal/i_components/ilowpass.icomp

```diff
@@ -18,16 +18,16 @@ pin in s32 in;
 pin out s32 out """\\fBout\\fR tracks \\fBin\\fR*\\fBscale\\fR through a low-pass
 filter of \\fBgain\\fR per period.""";
 
-param rw float scale = 1024 """A scale factor applied to the output
+pin io float scale = 1024 """A scale factor applied to the output
 value of the low-pass filter.""";
 
-param rw float gain = .5 """Together with the period, sets the rate at
+pin io float gain = .5 """Together with the period, sets the rate at
 which the output changes.  Useful range is between 0 and 1, with higher
 values causing the input value to be tracked more quickly.  For
 instance, a setting of 0.9 causes the output value to go 90% of the way
 towards the input value in each period""";
 
-variable double value;
+variable hal_float_t value;
```

Comment:
Indeed!
As both values multiplied are hal_float_t and I assume the only purpose of casting it is to round off to an int, I will just use (hal_s32_t)
I have no idea why __64 was used